lua/hop/hint.lua

```diff
@@ -20,36 +20,63 @@ end
 -- Regex hint mode.
 --
 -- Used to hint result of a search.
-function M.by_searching(pat, plain_search)
+function M.by_searching(pat, plain_search, oneshot)
   if plain_search then
     pat = vim.fn.escape(pat, '\\/.$^~[]')
   end
+  local re = vim.regex(pat)
   return {
-    oneshot = false,
+    oneshot = oneshot,
     match = function(s)
-      return vim.regex(pat):match_str(s)
+      return re:match_str(s)
     end
   }
 end
 
 -- Wrapper over M.by_searching to add support for case sensitivity.
 function M.by_case_searching(pat, plain_search, opts)
+  local ori = pat
   if plain_search then
     pat = vim.fn.escape(pat, '\\/.$^~[]')
   end
 
+  -- append dict pattern for each char in `pat`
+  local dict_pat = ''
+  for i = 1, #ori do
+    local char = ori:sub(i, i)
+    local dict_char_pat = ''
+    -- checkout dict-char pattern from each dict
+    for _, v in ipairs(opts.dict_list) do
+      local val = require('hop.dict.' .. v)[char]
```

Comment:
I’m not sure how caching will be done by the Lua runtime here but even then I’m not liking this style, especially because that loop could be partially moved out to prevent having to get the cached dict.

Reply:
How about executing dict `require` at setup()?
[This commit](https://github.com/yehuohan/hop.nvim/commit/0bca4af3f3995b0695e1e18d4446d9a8fbf05070) is a simple fix to this issue. If it's a correct fix, I'm pleasure to merge it to this PR.